src/hooks/useIsMounted/useIsMounted.ts

```diff
@@ -0,0 +1,25 @@
+import { type RefObject, useEffect, useRef } from 'react';
+
+/**
+ * Keeps track of whether the component is mounted, and returns a ref object that is updated
+ * accordingly.
+ *
+ * - `isMounted.current` is `false` before the component is mounted
+ * - `isMounted.current` is `true` after the component is mounted
+ * - `isMounted.current` is `false` during the unmount cleanup phase and after the component is
+ *   unmounted
+ *
+ * @returns A boolean ref object that reflects whether the component is mounted.
+ */
+export function useIsMounted(): RefObject<boolean> {
+  const isMounted = useRef(false);
+
+  useEffect(() => {
+    isMounted.current = true;
```

Comment:
Shouldn't the mounted state be updated after the mount is complete?

```ts
useEffect(() => {
  queueMicrotask(() => {
    isMounted.current = true;
  });

  ...
```

Reply:
I guess it depends on when we consider something mounted.

I would probably say; after the component has done it's first "render pass", and it exists in the DOM, and all refs have been set. So a soon as useEffects are called, those happen after mount.

So the DOM is available, so the rendering happened, so mounting is complete.

Although technically the mounting is still busy until `useLayoutEffect` is executed, but we should probably not change it to that?

I guess it depends on how it's used.

If other `useEffect`s are executed, what would the `isMounted.current` expected to be?